File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    cnt = 0
    
    tangerine_dict = Counter(tangerine)
    tangerine = sorted(tangerine_dict.items(), key=lambda x: x[1], reverse=True)
    
    while cnt < k:
        cnt += tangerine[answer][1]
        answer += 1
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int sum = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int t: tangerine) {
            map.put(t, map.getOrDefault(t, 0)+1);
        }
        List<Integer> cnt = new ArrayList<>(map.values());
        cnt.sort(Collections.reverseOrder());
        
        for (int c: cnt) {
            if (sum < k) {
                sum += c;
                answer++;
                }
            else {
                return answer;
            }
        }
        return answer;
    }
}